### PR TITLE
create cart page/functionality

### PR DESCRIPTION
**Pull request by nikitonskii**

### CODE REVIEW

File: src/components/CartItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import CartItem from "./CartItem";
```
Comment (twixDuo) [inline src/components/CartItem/index.tsx:1]: I think better will be, if you ```import``` change to ```export default``` for your component

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -28,8 +32,15 @@ const Header: React.FC = (): JSX.Element => {
           <button className="header-icon-button" onClick={logOut}>
             <i className="fas fa-sign-out-alt header-icon"></i>
           </button>
-          <button className="header-icon-button">
+          <button className="header-icon-button" onClick={moveToCart}>
             <i className="fas fa-shopping-cart header-icon"></i>
+            {addedProducts.length && (
```
Comment (twixDuo) [inline src/components/Header/Header.tsx]: What you think if we create const with name ```const isEmptyCart = addedProducts.length``` and then we will take name this const

File: src/pages/productsList/ProductsList.tsx [hunk 3]
```diff
@@ -38,7 +38,14 @@ const ProductsList: React.FC = (): JSX.Element => {
             } else {
               return (
                 <li className="product-list-wrapper" key={product.created}>
-                  <ProductCard {...product} />
+                  <ProductCard
```
Comment (twixDuo) [inline src/pages/productsList/ProductsList.tsx]: Why you repeat this code ? and add only one new key ```cargo_capacity``` What do you think, can you do without repeat this code ?

File: src/store/reducers/cartReducer/index.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      return {
+        ...state,
+        addedProducts: [...state.addedProducts, action.product],
+      };
+    case actionTypes.REMOVE_FROM_CART:
+      return {
+        ...state,
+        addedProducts: [
+          ...state.addedProducts.filter(
+            (item) => item.created !== action.created,
```
Comment (twixDuo) [inline src/store/reducers/cartReducer/index.ts]: Why you don't add type for item ?

Comment (twixDuo) [reply src/store/reducers/cartReducer/index.ts]: And why you use spred ? if filter return new array, and you can write 
```
addedProducts: state.addedProducts.filter((item) => item.created !== action.created)
```

File: src/types/products.ts [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
-  length: string;
-  max_atmosphering_speed: string;
-  crew: string;
-  passengers: string;
-  cargo_capacity: string;
-  consumables: string;
-  vehicle_class: string;
-  pilots: string[];
-  films: string[];
+  name?: string;
```
Comment (twixDuo) [inline src/types/products.ts]: why all keys with ```?``` ?

**Review by twixDuo — CHANGES_REQUESTED** [review]

File: src/components/CartItem/CartItem.tsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+const CartItem: React.FC<ProductItem> = ({
+  name,
+  model,
+  cost_in_credits,
+  created,
+}): JSX.Element => {
+  const dispatch = useDispatch();
+
+  const deleteProduct = (id: string | number) =>
```
Comment (twixDuo) [inline src/components/CartItem/CartItem.tsx]: Why ```id``` has type ```string | number``` ?

File: src/components/CartItem/CartItem.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    <div className="cart-item-container">
+      <h3>You have ordered :{name}</h3>
+      <p>
+        <b>Model :</b> {model}
+      </p>
+      <p>
+        <b>Price :</b>
+        {cost_in_credits}
+      </p>
+      <button onClick={() => deleteProduct(created)}>Delete from cart</button>
```
Comment (twixDuo) [inline src/components/CartItem/CartItem.tsx:36]: Here you pass key ```create```, but this function wait ```id```, where is logic ?)

**Review by twixDuo — CHANGES_REQUESTED** [review]

File: src/components/CartItem/index.tsx (hunk 1 above)
Comment (nikitonskii) [reply src/components/CartItem/index.tsx:1]:
```ts
const a: any = 'Hello world'
```

**Pull request merged**